nullaway/src/main/java/com/uber/nullaway/NullAway.java

```diff
@@ -399,16 +400,31 @@ public Description matchMethodInvocation(MethodInvocationTree tree, VisitorState
     if (!withinAnnotatedCode(state)) {
       return Description.NO_MATCH;
     }
-    final Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
-    if (methodSymbol == null) {
-      throw new RuntimeException("not expecting unresolved method here");
-    }
+    Symbol.MethodSymbol methodSymbol = getSymbolForMethodInvocation(tree, state);
     handler.onMatchMethodInvocation(this, tree, state, methodSymbol);
     // assuming this list does not include the receiver
     List<? extends ExpressionTree> actualParams = tree.getArguments();
     return handleInvocation(tree, state, methodSymbol, actualParams);
   }
 
+  private static Symbol.MethodSymbol getSymbolForMethodInvocation(
+      MethodInvocationTree tree, VisitorState state) {
+    Symbol.MethodSymbol methodSymbol = ASTHelpers.getSymbol(tree);
+    Verify.verify(methodSymbol != null, "not expecting unresolved method here");
+    // For interface methods, if the method is an implicit method corresponding to a method from
+    // java.lang.Object, use the symbol for the java.lang.Object method instead.  We do this to
+    // properly treat the method as unannotated, which is particularly important for equals()
+    // methods.  This is an adaptation to a change in JDK 18; see
+    // https://bugs.openjdk.org/browse/JDK-8272564
+    if (methodSymbol.owner.isInterface()) {
+      Symbol.MethodSymbol baseSymbol = (Symbol.MethodSymbol) methodSymbol.baseSymbol();
+      if (baseSymbol != methodSymbol && baseSymbol.owner == state.getSymtab().objectType.tsym) {
+        methodSymbol = baseSymbol;
+      }
+    }
+    return methodSymbol;
+  }
```

Comment:
Correct me if I'm understanding this wrong, but previously we would get a symbol from the interface, which would eventually be considered annotated in the caller (`matchMethodInvocation` and later `handleInvocation`).

Now that we have this special handling to resolve to the base symbol (i.e., from `java.lang.Object`), which would be considered unannotated (everything from the JDK is unannotated), and then NullAway won't think that `equals` must have a nonnull param. Right?

If so, this change looks good to me! (I also agree that this is the interim solution and the end goal is for NullAway to enforce nullable on `equals`, but that's for another PR)

Reply:
@yuxincs yes, this is exactly correct.